### ISSUE TITLE
Setting Facets on Slaves

### ISSUE THREAD

**Issue by robertjwhitney** [algolia/algoliasearch-rails#186]
Hi just a sanity check, in the DSL, can I set `attributesForFaceting` on slaves? I have them set for each slave, v 1.13.0,  and they don't seem to be propagating to the slave indexes...

**Comment by robertjwhitney** [comment]
Also, I should mention that I'm using `add_slave` inside of an `add_index` block here

**Comment by robertjwhitney** [comment]
Just digging around a little bit, should something be done with the `block` here? https://github.com/algolia/algoliasearch-rails/blob/2d63c37ed6151e314e5479e6a102d1f48e23cd4f/lib/algoliasearch-rails.rb#L246

**Comment by robertjwhitney** [comment]
I did a little more digging yesterday, seems like the settings for the slaves of added indexes never get applied when they're all getting looped through... will have to do some more source diving to track down where that might need to change.

**Comment by redox** [comment]
> Hi just a sanity check, in the DSL, it looks like when I add_slave inside of an add_index block, all the settings are ignored.

Arg indeed, I don't think this is supported :/ We should patch the DSL to forbid it.

Does it mean you're targeting several indices (with slaves) from a single model @robertjwhitney ?

**Comment by robertjwhitney** [comment]
@redox yep exactly. I need to right now for legacy support on some clients, but would like to migrate away eventually.

**Comment by robertjwhitney** [comment]
Is there a good way we could add support? Seems like a pretty valid use-case?

**Comment by redox** [comment]
> Is there a good way we could add support? Seems like a pretty valid use-case?

I need to check whether this has a big impact or not; the current `replica` handling is a bit messy and would eventually need to a full rewrite (gem was designed when replicas were not introduced yet).

I'm not confident providing any ETA; and I would recommend going for custom `ruby` code to set the settings & use the index.

**Comment by robertjwhitney** [comment]
Cool, well, I'm happy to help it along if you come up with a plan for doing that. Replica's are a pretty important feature, and we'd love to have full support!
